templates/maps/google-maps-with-search.php

```diff
@@ -120,5 +125,39 @@ function autoCompleteAddress(){
                 }
             });
         }
+
+        function reverseGeocode( latLng ) {
+            let data = {
+                location: {
+                    lat: latLng.lat(),
+                    lng: latLng.lng(),
+                }
+            };
+
+            geocoder.geocode( data )
+                .then( function ( { results: addresses } ) {
+                    let address = addresses[0].address_components
```

Comment:
Isn't that necessary to check `addresses` length > 0 or not.
as you're using `addresses[0]`, hope, it would be better to check if `addresses.length > 0` then do rest.